crates/nu-command/src/strings/str_/upcase.rs

```diff
@@ -0,0 +1,109 @@
+use nu_engine::CallExt;
+use nu_protocol::ast::Call;
+use nu_protocol::ast::CellPath;
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value};
+
+#[derive(Clone)]
+pub struct SubCommand;
+
+impl Command for SubCommand {
+    fn name(&self) -> &str {
+        "str upcase"
+    }
+
+    fn signature(&self) -> Signature {
+        Signature::build("str upcase").rest(
+            "rest",
+            SyntaxShape::CellPath,
+            "optionally upcase text by column paths",
+        )
+    }
+
+    fn usage(&self) -> &str {
+        "upcases text"
+    }
+
+    fn run(
+        &self,
+        engine_state: &EngineState,
+        stack: &mut Stack,
+        call: &Call,
+        input: PipelineData,
+    ) -> Result<PipelineData, ShellError> {
+        operate(engine_state, stack, call, input)
+    }
+
+    fn examples(&self) -> Vec<Example> {
+        vec![Example {
+            description: "Upcase contents",
+            example: "'nu' | str upcase",
+            result: Some(Value::test_string("NU")),
+        }]
+    }
+}
+
+fn operate(
+    engine_state: &EngineState,
+    stack: &mut Stack,
+    call: &Call,
+    input: PipelineData,
+) -> Result<PipelineData, ShellError> {
+    let head = call.head;
+    let column_paths: Vec<CellPath> = call.rest(engine_state, stack, 0)?;
+    input.map(
+        move |v| {
+            if column_paths.is_empty() {
+                action(&v, head)
+            } else {
+                let mut ret = v;
+                for path in &column_paths {
+                    let r =
+                        ret.update_cell_path(&path.members, Box::new(move |old| action(old, head)));
+                    if let Err(error) = r {
+                        return Value::Error { error };
+                    }
+                }
+                ret
+            }
+        },
+        engine_state.ctrlc.clone(),
+    )
+}
+
+fn action(input: &Value, head: Span) -> Value {
+    match input {
+        Value::String { val: s, .. } => Value::String {
+            val: s.to_ascii_uppercase(),
```

Comment:
I wonder... should this be `to_uppercase` instead of `to_ascii_uppercase`? Nushell can work with utf-8 strings, and that'd let it also uppercase more non-English words if I understand the docs right

Reply:
good point. I think you're right, we probably should use `to_uppercase`.